streams/fixed_chunk_stream.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * An interface specifying the requirements for the {@linkcode FixedChunkStream} to
+ * successfully resize the chunks, with the exception of the last chunk.
+ */
+export interface Fixable<T, U = T extends ArrayLike<infer V> ? V : never>
+  extends ArrayLike<U> {
+  /**
+   * Validates that a constructor property exists on the type. The constructor
+   * needs a signature of `new(length: number): T`, but TypeScript is unable to
+   * validate this, so you must.
+   */
+  constructor: Exclude<unknown, undefined>;
+  /**
+   * A method that inserts the `ArrayLike<U>` in `T` at the optional `offset`
+   * index.
+   */
+  set(array: ArrayLike<U>, offset?: number): void;
+  /**
+   * A method that slices a chunk from `start` to `end` returning `T`
+   */
+  slice(start?: number, end?: number): T;
+  /**
+   * A property that returns the length of `T`
+   */
+  readonly length: number;
+}
+
+/**
+ * A transform stream that resize the chunks into perfectly `size` chunks with
+ * the exception of the last chunk.
+ *
+ * Works on any type that meets the {@linkcode Fixable} requirements.
+ * TypeScript is unable to validate the constructor requirement of needing a
+ * signature of `new(length:number): T` so you must validate this yourself.
+ *
+ * @typeParam T is the type inside the stream. `Uint8Array` for example.
+ *
+ * @example Usage
+ * ```ts
+ * import { FixedChunkStream } from "@std/streams/fixed-chunk-stream";
+ * import { assertEquals } from "@std/assert/equals";
+ *
+ * const readable = ReadableStream.from(function* () {
+ *   let count = 0
+ *   for (let i = 0; i < 100; ++i) {
+ *     const array = new Uint8Array(Math.floor(Math.random() * 1000));
+ *     count += array.length;
+ *     yield array;
+ *   }
+ *   yield new Uint8Array(512 - count % 512)
+ * }())
+ *   .pipeThrough(new FixedChunkStream(512))
+ *   .pipeTo(new WritableStream({
+ *     write(chunk, _controller) {
+ *       assertEquals(chunk.length, 512)
+ *     }
+ *   }))
+ * ```
+ *
+ * > [!WARNING]
+ * > **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @experimental
+ */
+export class FixedChunkStream<T extends Fixable<T>>
+  extends TransformStream<T, T> {
```

Comment:
I understand that you'd like to have the input be very flexible. However, it would make this the only API in `@std/streams` of this kind. Should demand for this functionality arise in the future, it should be something that applied to all of `@std/streams`, where applicable, and can be done in a non-breaking manner. Let's also remove `Fixable`.
```suggestion
export class FixedChunkStream extends TransformStream<Uint8Array, Uint8Array> {
```